src/box/relay_stream.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2024, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include "iostream.h"
+#include "small/lsregion.h"
+#include "xrow.h"
+#include "memory.h"
+
+/** Written data size after which the stream should be flushed. */
+extern size_t relay_stream_flush_size;
+
+/**
+ * A structure encapsulating writes made by relay. Collects the rows into a
+ * buffer and flushes it to the network as soon as its size reaches a specific
+ * threshold.
+ */
+struct relay_stream {
```

Comment:
I don't think that "stream" is a proper name. It's more like a buffer.

Reply:
It will become a stream as a part of #9968. Data will be written into it potentially with a yield, with no explicit flush required. From multiple places in relay. And one fiber will be handling the stream to actually forward the data into a socket (which will probably be inside this struct). It is called here "stream" so as we wouldn't need to rename it later.

Reply:
> And one fiber will be handling the stream to actually forward the data into a socket (which will probably be inside this struct)

Yes, during some WIP iterations I tried to hide the `struct iostream` from relay into the `relay_stream`. So that it would have both `relay_stream_read` and `relay_stream_write` methods. And `relay_stream_read` would simply call `coio_read_xrow`. But I encountered some problems with needing to write to `iostream` from tx (some initial relay messages are sent from there), and then initializing the stream in relay. Although I think this may be bypassed. IIRC, `lsregion` doesn't have to be used in the same thread it was initialized in.